src/app/stores/app_store.ts

```diff
@@ -1070,6 +1072,34 @@ export class AppStore extends BaseStore {
     }
   }
 
+  /**
+   * Updates all scales values
+   */
+  public updateScales() {
+    try {
+      this.chart.scales.forEach(scale => {
+        const mappings = [...this.chart.elements, ...this.chart.glyphs.flatMap(gl => gl.marks)].flatMap(el => {
```

Comment:
This line is really compact, can we make it more legible? Maybe with formatting, or maybe with temp variables.